apps/desktop/src/lib/branches/branchListing.ts

```diff
@@ -228,35 +228,42 @@ export class CombinedBranchListingService {
 	}
 
 	search(searchTerm: Readable<string | undefined>) {
-		return derived(
-			[searchTerm, this.combinedSidebarEntries],
-			([searchTerm, combinedSidebarEntries]) => {
-				if (!searchTerm) return [];
-
-				const fuse = new Fuse(combinedSidebarEntries, {
-					keys: [
-						// Subject is branch listing
-						'subject.name',
-						'subject.lastCommiter.email',
-						'subject.lastCommiter.name',
-						// Subject is pull request
-						'subject.title',
-						'subject.author.email',
-						'subject.author.name'
-					],
-					threshold: 0.3, // 0 is the strictest.
-					sortFn: (a, b) => {
-						// Sort results by when the item was last modified.
-						const dateA = (a.item.modifiedAt || a.item.updatedAt) as Date | undefined;
-						const dateB = (b.item.modifiedAt || b.item.updatedAt) as Date | undefined;
-						if (dateA && dateB) {
-							return dateA < dateB ? -1 : 1;
-						}
-						return 0;
+		const engine = derived([this.combinedSidebarEntries], ([combinedSidebarEntries]) => {
+			const fuse = new Fuse(combinedSidebarEntries, {
+				keys: [
+					// Subject is branch listing
+					'subject.name',
+					'subject.lastCommiter.email',
+					'subject.lastCommiter.name',
+					// Subject is pull request
+					'subject.title',
+					'subject.author.email',
+					'subject.author.name'
+				],
+				threshold: 0.3, // 0 is the strictest.
+				ignoreLocation: true,
+				isCaseSensitive: false,
+				sortFn: (a, b) => {
+					// Sort results by when the item was last modified.
+					const dateA = (a.item.modifiedAt ?? a.item.updatedAt) as Date | undefined;
+					const dateB = (b.item.modifiedAt ?? b.item.updatedAt) as Date | undefined;
+					if (dateA !== undefined && dateB !== undefined && dateA !== dateB) {
+						return dateA < dateB ? -1 : 1;
 					}
-				});
+					// If there are no dates or they're the same, sort by score
+					return a.score < b.score ? -1 : 1;
+				}
+			});
 
-				return fuse.search(searchTerm, { limit: 100 }).map((result) => result.item);
+			return fuse;
```

Comment:
Can you return this immediately instead of declaring variable -> return variable?